### PR TITLE
#43 - Added expiration claim method

### LINKED ISSUE #43: Create an expiration based claim method
## Description:
Create a new expiration-based claim method

- The claim method will take an expiration time stamp
- Allow the beneficiary to claim before the expiration time stamp
- Write new tests for the method

### PR DESCRIPTION
**Pull request by AnkithVijay**
Create a new expiration-based claim method
• The claim method will take an expiration time stamp 
• Allow the beneficiary to claim before the expiration time stamp 
• Write new tests for the method

### CODE REVIEW

File: contracts/SafientMain.sol [hunk 1]
```diff
@@ -40,25 +40,22 @@ contract SafientMain is Safes, Claims, Guardians, IArbitrable {
...
+     * @param _claimValue Value of claim is uint256 it can be signaling period or dday or eday
      * @param _metaEvidence URL of the metaevidence
      */
     function createSafe(
         address _beneficiary,
         string memory _safeId,
         Types.ClaimType _claimType,
-        uint256 _signalingPeriod,
-        uint256 _DDay,
+        uint256 _claimValue,
```
Comment (koshikraj) [inline contracts/SafientMain.sol]: Let's rename it to _claimPeriod

File: contracts/components/Claims.sol [hunk 2]
```diff
@@ -235,17 +254,43 @@ contract Claims {
...
+     * @notice Create a new Expiry-Day based claim
+     * @param _safeId Id of the safe
+     * @param data Includes safe data
+     */
+    function _createExpirionBasedClaim(
+        string memory _safeId,
+        Types.ExpirionClaimData memory data
+    ) internal ExpirionBasedClaim(_safeId, data) {
+        claimsCount += 1;
+        require(block.timestamp < data.expiryDay, "Claim has been expired");
```
Comment (koshikraj) [inline contracts/components/Claims.sol]: Change it to `Safe has been expired`

File: contracts/components/Safes.sol [hunk 3]
```diff
@@ -158,11 +168,9 @@ contract Safes {
             createdBy: msg.sender,
             currentOwner: msg.sender,
             beneficiary: _beneficiary,
-            signalingPeriod: _signalingPeriod,
-            endSignalTime: 0,
-            latestSignalTime: 0,
-            dDay: _DDay,
+            claimValue: _claimValue,
```
Comment (koshikraj) [inline contracts/components/Safes.sol]: Let's rename this to `claimPeriod`

File: contracts/libraries/Types.sol [hunk 4]
```diff
@@ -40,7 +39,7 @@ library Types {
     struct Claim {
         uint256 id;
         address claimedBy;
-        ClaimType claimType;
+        ClaimType claimType;// need to remove
```
Comment (koshikraj) [inline contracts/libraries/Types.sol]: We can remove this if not used.

Comment (AnkithVijay) [reply contracts/libraries/Types.sol]: we can remove it, but we are not storing any safe id in the Claims for reference of claim type, if we don't need it we can remove

Comment (koshikraj) [reply contracts/libraries/Types.sol]: We can remove it as we store the claim index on Safe and also we will be storing all the claims on our Textile threadDb store.

**Review by koshikraj — APPROVED** [review]

**Pull request merged**